spinoso-time/src/time/tzrs/offset.rs

```diff
@@ -0,0 +1,273 @@
+            "F" => Self::fixed(6),
+            "G" => Self::fixed(7),
+            "H" => Self::fixed(8),
+            "I" => Self::fixed(9),
+            "K" => Self::fixed(10),
+            "L" => Self::fixed(11),
+            "M" => Self::fixed(12),
+            "N" => Self::fixed(-1),
+            "O" => Self::fixed(-2),
+            "P" => Self::fixed(-3),
+            "Q" => Self::fixed(-4),
+            "R" => Self::fixed(-5),
+            "S" => Self::fixed(-6),
+            "T" => Self::fixed(-7),
+            "U" => Self::fixed(-8),
+            "V" => Self::fixed(-9),
+            "W" => Self::fixed(-10),
+            "X" => Self::fixed(-11),
+            "Y" => Self::fixed(-12),
+            "Z" => Self::utc(),
+            _ => {
+                lazy_static! {
+                    static ref HH_MM_MATCHER: Regex = Regex::new(r"^([\-\+]{1})(\d{2})(\d{2})$").unwrap();
+                }
+                if HH_MM_MATCHER.is_match(input) {
+                    let caps = HH_MM_MATCHER.captures(input).unwrap();
+
+                    let sign = if caps.get(1).unwrap().as_str() == "+" { 1 } else { -1 };
+                    let hours = caps.get(2).unwrap().as_str().parse::<i32>().unwrap();
+                    let minutes = caps.get(3).unwrap().as_str().parse::<i32>().unwrap();
+
+                    let offset_seconds: i32 = sign * ((hours * SECONDS_IN_HOUR) + (minutes * SECONDS_IN_MINUTE));
+                    Self::fixed(offset_seconds)
+                } else {
+                    // TODO: ArgumentError
+                    Self::utc()
+                }
+            }
+        }
+    }
+}
+
+impl From<String> for Offset {
+    fn from(input: String) -> Self {
+        Offset::from(input.as_str())
+    }
+}
+
+impl From<TimeZoneRef<'static>> for Offset {
+    #[inline]
+    #[must_use]
+    fn from(tz: TimeZoneRef<'static>) -> Self {
+        Self::tz(tz)
+    }
+}
+
+impl From<i32> for Offset {
+    /// Construct a Offset with the offset in seconds from UTC
+    #[inline]
+    #[must_use]
+    fn from(seconds: i32) -> Self {
+        Self::fixed(seconds)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn offset_seconds_from_fixed_offset(input: &str) -> i32 {
+        let offset = Offset::from(input);
+        let local_time_type = offset.time_zone_ref().local_time_types()[0];
+        local_time_type.ut_offset()
+    }
+
+    fn offset_name(offset: &Offset) -> &str {
+        match offset {
+            Offset::Utc => "UTC",
+            Offset::Fixed(ltt) => ltt[0].time_zone_designation(),
+            Offset::Tz(_) => "Ambiguous timezone name",
+        }
+    }
+
+    #[test]
+    fn fixed_zero_is_not_utc() {
+        let offset = Offset::from(0);
+        assert!(matches!(offset, Offset::Fixed(_)));
+    }
+
+    #[test]
+    fn z_is_utc() {
+        let offset = Offset::from("Z");
+        assert!(matches!(offset, Offset::Utc));
+    }
+
+    #[test]
+    fn from_str_hh_mm() {
```

Comment:
let's add a new test method that's basically a duplicate of this but with the colon separators.